sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/BeamSqlDslUdfUdafTest.java

```diff
@@ -133,6 +137,40 @@ public void testUdf() throws Exception {
     pipeline.run().waitUntilFinish();
   }
 
+  /** test auto-registered UDF/UDAF. */
+  @Test
+  public void testAutoUdfUdaf() throws Exception {
+    Schema resultType =
+        Schema.builder().addInt32Field("f_int2").addInt32Field("squarecubicsum").build();
+
+    Row row = Row.withSchema(resultType).addValues(0, 4890).build();
+
+    String sql =
+        "SELECT f_int2, squaresum(cubic(f_int)) AS `squarecubicsum`"
+            + " FROM PCOLLECTION GROUP BY f_int2";
+    PCollection<Row> result =
+        boundedInput1.apply("testUdaf", SqlTransform.query(sql).autoRegisterUdfUdaf());
+
+    PAssert.that(result).containsInAnyOrder(row);
+    pipeline.run().waitUntilFinish();
+  }
+
+  /** Auto register for test. */
+  @AutoService(UdfUdafRegister.class)
+  public static class UdfUdafRegisterTest implements UdfUdafRegister {
+
+    @Override
+    public Map<String, Class<? extends BeamSqlUdf>> getEvalUdfs() {
+      return new HashMap<String, Class<? extends BeamSqlUdf>>(
+          ImmutableMap.of("cubic", CubicInteger.class));
```

Comment:
These udfs are registered as long as the test jar is in the classpath. Can you make the name of these test functions a little more unique so we don't have collisions?